### ISSUE TITLE
docdb: Group All TSDescriptor Setters Into One Method

### ISSUE THREAD

**Issue by rahuldesirazu** [yugabyte/yugabyte-db#7571]
On each heartbeat, master calls TSDescriptor methods like 
```
  void set_num_sst_files (uint64_t num_sst_files) {
    std::lock_guard<decltype(lock_)> l(lock_);
    ts_metrics_.num_sst_files = num_sst_files;
  }
```

that takes a write lock, sets a field, and releases the write lock. To avoid doing this many times over for each individual field, create a method that takes a lock once and sets every field before releasing the lock.

**Comment by jackvstrickland** [comment]
I'd like to jump in on this one. Can you point me to where the master is calling these methods?

**Comment by bmatican** [comment]
Hey @jackvstrickland , thanks for the interest in Yugabyte!

After doing some more checking, I think we might already have this, in the form of `TSDescriptor::UpdateMetrics`. Perhaps one nice cleanup could be to try to remove all the individual setters from the TSDescriptor class and confirm nothing breaks! If you'd be interested in that, give it a shot and I can pass the issue to you!

If you'd like some more real time help, or help with other good first issues, feel free to join our slack channel and hop onto the contributors channel!

**Comment by bozhiyou** [comment]
> Perhaps one nice cleanup could be to try to remove all the individual setters from the TSDescriptor class and confirm nothing breaks!

Is this still available? I'd like to jump in if not.

**Comment by jackvstrickland** [comment]
I haven't started, in fact I just saw this. go ahead!

**Comment by bmatican** [comment]
@bozhiyou go for it, if @jackvstrickland is not going to! let me know if you need anymore input!

**Comment by bozhiyou** [comment]
Note that this issue also existed during heartbeat updates. Fixed in #8357 as `TSDescriptor::UpdateHeartbeat` in addition to `TSDescriptor::UpdateMetrics`.
